NebulaPatcher/Patches/Transpilers/UIProductionStatWindow_Patch.cs

```diff
@@ -0,0 +1,140 @@
+                    codes[i + 9] = new CodeInstruction(OpCodes.Stloc_S, 29);
+                }
+
+                //Patch fix for the "Picking specific planet" in statistics
+                if (codes[i].operand?.ToString() == "PowerSystem powerSystem" && patchActive[1] && !patchActive[2])
+                {
+                    patchActive[2] = true;
+                    for (int j = 0; j < patchLength[2]; j++)
+                    {
+                        codes[i - 2 + j].opcode = OpCodes.Nop;
+                    }
+                    codes[i] = new CodeInstruction(OpCodes.Ldc_I4_0);
+                    codes[i + 1] = new CodeInstruction(OpCodes.Conv_I8);
+                    codes[i + 2] = new CodeInstruction(OpCodes.Stloc_S, 40);
+                    codes[i + 3] = new CodeInstruction(OpCodes.Ldloca_S, 40);
+                    codes[i + 4] = new CodeInstruction(OpCodes.Ldarg_0);
+                    codes[i + 5] = new CodeInstruction(OpCodes.Ldfld, targetIndex);
+                    codes[i + 6] = new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(StatisticsManager), "UpdateTotalChargedEnergy"));
+                    codes[i + 7] = new CodeInstruction(OpCodes.Ldloc_S, 33);
+                    codes[i + 8] = new CodeInstruction(OpCodes.Ldfld, typeof(FactoryProductionStat).GetField("energyConsumption", BindingFlags.Public | BindingFlags.Instance));
+                    codes[i + 9] = new CodeInstruction(OpCodes.Stloc_S, 41);
+                }
+              
+                //Patch fix for the "Picking specific star system" in statistics
+                if (codes[i].operand?.ToString() == "PowerSystem powerSystem" && patchActive[2] && !patchActive[3])
+                {
+                    patchActive[3] = true;
+                    for (int j = 0; j < patchLength[3]; j++)
+                    {
+                        codes[i - 2 + j].opcode = OpCodes.Nop;
+                    }
+                    codes[i] = new CodeInstruction(OpCodes.Ldloca_S, 47); //Loads the address of the local variable at a specific index onto the evaluation stack, short form.
+                    codes[i + 1] = new CodeInstruction(OpCodes.Ldarg_0);
+                    codes[i + 2] = new CodeInstruction(OpCodes.Ldfld, targetIndex);
+                    codes[i + 3] = new CodeInstruction(OpCodes.Call, AccessTools.Method(typeof(StatisticsManager), "UpdateTotalChargedEnergy"));
+                }   
+            }
+            //return codes.AsEnumerable();
+            return ReplaceFactoryCondition(codes);
+        }
+
+        public static IEnumerable<CodeInstruction> ReplaceFactoryCondition(IEnumerable<CodeInstruction> instructions)
+        {
+            //change: if (starData.planets[j].factory != null)
+            //to    : if (starData.planets[j].factoryIndex != -1)
```

Comment:
I really like the way you formatted this comment, could you add more comments like this to the changes that `UpdatePower_Transpiler` does under the hood. It would really help with maintainability.

Reply:
I have added a comment there about it.